src/Microsoft.ML.Dnn/ImageClassificationTransform.cs

```diff
@@ -749,6 +765,10 @@ private protected override void SaveModel(ModelSaveContext ctx)
             ctx.Writer.Write(_predictedLabelColumnName);
             ctx.Writer.Write(_learningRate);
             ctx.Writer.Write(_classCount);
+
+            for (int j = 0; j < _classCount; j++)
```

Comment:
>_classCount [](start = 32, length = 11)

Can you add an assert here that the length of `_keyValueAnnotations` is equal to `_classCount` or that it is null?